File: 121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size() == 0 || prices.size() == 1)return 0;
        int mm = prices[0];
        int sol = 0;
        for(int i = 1; i < prices.size(); i++){
            sol = max(sol, prices[i]-mm);
            mm = min(mm, prices[i]);
        }
        return sol;
    }
};

File: Python/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        sol = 0
        mm = prices[0]
        for i in range(1, len(prices)):
            curr = prices[i] - mm
            sol = max(sol, curr)
            mm = min(prices[i], mm)
        return sol
